src/validation/index.js

```diff
@@ -0,0 +1,133 @@
+import { superstruct } from "superstruct";
+import isUuid from "is-uuid";
+import isEmail from "../utils/validate-rfc822-email";
+import { parsePhoneNumber, ParseError } from "libphonenumber-js";
+
+/**
+ * Cached compiled regexes for performance.
+ */
+const validISO8601 = /^([+-]?\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24:?00)([.,]\d+(?!:))?)?(\17[0-5]\d([.,]\d+)?)?([zZ]|([+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?(Z|[+-](?:2[0-3]|[01][0-9])(?::?(?:[0-5][0-9]))?)?$/;
+const validSSN = /^[\d]{3}-[\d]{2}-[\d]{4}$/;
+const validEIN = /^[\d]{2}-[\d]{7}$/;
+
+/**
+ *  Helpers
+ */
+const handleRequired = val =>
+  (val === undefined || val === "") && "missing_required";
+
+/**
+ *  Type validator functions.
+ *  Named after types, should be Capitalized.
+ */
+const AWSDate = value =>
+  handleRequired(value) || validISO8601.test(value) || "invalid_date";
+const AWSEmail = value =>
+  handleRequired(value) || isEmail(value) || "invalid_email";
+
+const AWSPhone = value => {
+  try {
+    const number = parsePhoneNumber(value);
```

Comment:
why the need to assign this to a var?

Reply:
Method chaining with this library does weird shit.  It's not ideal but I'd rather do something like this than have to worry about a bunch of weird errors.  Even almost all their own docs do something similar.